src/network/networkd-link.c

```diff
@@ -327,11 +343,40 @@ void link_update_operstate(Link *link) {
         else
                 operstate = LINK_OPERSTATE_OFF;
 
+        /* Even if the link does not have SLAVE flag, we will set it as 'enslaved' when it has no
+         * Address= setting and is joined to other netdevs. */
+        if (IN_SET(operstate, LINK_OPERSTATE_DEGRADED, LINK_OPERSTATE_CARRIER) &&
+            (link->flags & IFF_SLAVE || (link->enslaved && link->network && !link->network->static_addresses)))
+                operstate = LINK_OPERSTATE_ENSLAVED;
+
```

Comment:
```link->enslaved``` should be only be set by ```IFF_SLAVE```  flag with respect to kernel. We should not consider any other things on our own.